pyposeidon/utils/cast.py

```diff
@@ -362,7 +363,10 @@ def run(self, **kwargs):
         logger.debug("create restart file")
 
         # check for combine hotstart
-        hotout = int((self.sdate - self.rdate).total_seconds() / info["params"]["core"]["dt"])
+        if ihot == 2:
+            hotout = int((self.sdate - self.rdate).total_seconds() / info["params"]["core"]["dt"])
+        elif ihot == 1:
+            hotout = self.parameters["nhot_write"]
```

Comment:
You should (almost) always add an `else` clase to an `if/elif`. 

What if the user adds an invalid `ihot` value? Then `hotout` will not be defined and you will get a `NameError` which will confusing. 

if instead we have:
``` python
else:
    raise ValueError("Acceptable values for `ihot` are 1 and 2, not: %s", ihot)
```
it will be immediately apparent what the issue is.

For the record, ideally, we should be validating the user input as soon as we get it, but to do that properly will require a major restructure so let's keep it simple for now.